src/modality-specific-files/task-events.md

```diff
@@ -175,10 +175,49 @@ This can be done by using a `/stimuli` directory or by reference to a stimuli da
+
+The temporal annotations for time-varying stimuli can be stored in a separate file, `stimulus.tsv`, which would hold the annotations for timepoints of the time-varying stimulus similar to an `events.tsv` file. `stimulus.tsv` MUST follow the same format and cotain the same REQUIRED and OPTIONAL columns of the `events.tsv` file, and MAY be accompanied by a JSON file describing the columns in detail.
+
+An exmaplry `stimuli.tsv` file is shown below:
+
+| stim_file                            | type        | NSD_id | COCO_id | first_COCO_description                         | HED                                                                                                                                                                                            |
+| ------------------------------------ | ----------- | ------ | ------- | ---------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| nsd03050.png                         | still_image | 3050   | 262414  | a person standing on a surfboard riding a wave | (Foreground-view, ((Item-count/1, ((Human, Human-agent), Body, Male, Agent-trait/Adolescent)), (Play, (Item-count/1, Man-made-object)))), (Background-view, (Outdoors, Natural-feature/Ocean)) |
+| the_present_movie.mp4                | movie       | n/a    | n/a     | n/a                                            | Visual-presentation, Movie                                                                                                                                                                     |
+| the_present-LogLumRatio_stimulus`.tsv | annotation  | n/a    | n/a     | n/a                                            | n/a                                                                                                                                                                                            |
+
+An exemplary `events.rsv` file is shown below:
+
+| onset  | duration | value    | stim_file                         | HED                                                         |
+| ------ | -------- | -------- | --------------------------------- | ----------------------------------------------------------- |
+| 25.033 | 1.02     | nsd03050 | {nsd03050, first_COCO_description} | Sensory-event, Visual-presentation, (Image, {nsd03050.png}) |
+
+In this example, the `{nsd03050.jpg, first_COCO_description}` indicates to append or add the descriptions of the image under the *first_COCO_description* in the `stimuli.tsv` file to a column in the `events.tsv` file with the same name. Also, following the HED-specifications 3.2.0, the HED annotation `{nsd03050.jpg}` indicates inserting the HED annotation from the `HED` column in the `stimuli.tsv` file to the `events.tsv` file's `HED` column for that particular event row.
```

Comment:
I have two questions. 

First, would it make sense to simply require a description column in the stimuli.tsv file? Instead of adding the curly brackets mechanism, this description is just always taken from there. This would avoid adding additional syntax people have to learn. 

Second, for inserting the HED annotations into the events file, does this only work if the HED annotations are in the column? I think it should work with the json HED annotations as well. Only I can think of two possible ways it could work, either the compiled HED annotation is taken to the event file based on reference to the file name, or we could simply extend the 'namespace' of columns and also include the column names in the stimulus files. Basically, once the reference to the stimulus is made, if there is column 'color' in the stimulus file we can use {color} in the HED tag. I can forsee use cases for this if particular properties of the image are relevant for a condition is a specific experiment. We might run into issues if column names are not unique, so this either needs to be specified or we need a prefix.

Reply:
Thanks a lot for your fantastic comments. If I understood them correctly, it seems that the first and second comments are (in a way) opposing each other. In the first comment you suggest to design a predtermined column that would be automatically pulled from `stimuli.tsv` to `events.tsv`. But the second comment provide a great way to use HED curly brace method to pull multiple columns into the HED annotations of each row.

Please correct me if I am worng. I believe that the solution should be as inclusive as possible.